insonmnia/miner/salesman.go

```diff
@@ -0,0 +1,343 @@
+package miner
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/noxiouz/zapctx/ctxlog"
+	"github.com/pborman/uuid"
+	"github.com/sonm-io/core/blockchain"
+	"github.com/sonm-io/core/insonmnia/cgroups"
+	"github.com/sonm-io/core/insonmnia/hardware"
+	"github.com/sonm-io/core/insonmnia/matcher"
+	"github.com/sonm-io/core/insonmnia/resource"
+	"github.com/sonm-io/core/insonmnia/state"
+	"github.com/sonm-io/core/proto"
+	"github.com/sonm-io/core/util"
+	"go.uber.org/zap"
+)
+
+type Salesman struct {
+	state         *state.Storage
+	resources     *resource.Scheduler
+	hardware      *hardware.Hardware
+	eth           blockchain.API
+	dwh           sonm.DWHClient
+	controlGroup  cgroups.CGroup
+	cGroupManager cgroups.CGroupManager
+	ethkey        *ecdsa.PrivateKey
+
+	askPlanCGroups map[string]cgroups.CGroup
+
+	matcher  matcher.Matcher
+	log      *zap.SugaredLogger
+	dealChan chan *sonm.Deal
+	mu       sync.Mutex
+}
+
+func NewSalesman(
+	ctx context.Context,
+	state *state.Storage,
+	resources *resource.Scheduler,
+	hardware *hardware.Hardware,
+	eth blockchain.API,
+	matcher matcher.Matcher,
+	ethkey *ecdsa.PrivateKey,
+) (*Salesman, error) {
```

Comment:
this is awful, maybe will be better to pass config structure or use options?

Reply:
i think that a config structure is better in terms of readability, but nevermind